podpac/core/algorithm/test/test_generic.py

```diff
@@ -28,80 +32,89 @@ def test_serialization(self):
-        node = Arithmetic(A=sine_node, B=sine_node, eqn="2*abs(A) - B + {offset}", params={"offset": 1})
-        output = node.eval(coords)
+            coords = podpac.Coordinates(
+                [podpac.crange(-90, 90, 1.0), podpac.crange(-180, 180, 1.0)], dims=["lat", "lon"]
+            )
+            sine_node = SinCoords()
+            node = Arithmetic(A=sine_node, B=sine_node, eqn="2*abs(A) - B + {offset}", params={"offset": 1})
+            output = node.eval(coords)
 
-        a = sine_node.eval(coords)
-        b = sine_node.eval(coords)
-        np.testing.assert_allclose(output, 2 * abs(a) - b + 1)
+            a = sine_node.eval(coords)
+            b = sine_node.eval(coords)
+            np.testing.assert_allclose(output, 2 * abs(a) - b + 1)
 
     def test_evaluate_not_allowed(self):
-        podpac.settings.set_allow_python_eval_exec(False)
+        with podpac.settings:
+            podpac.settings.set_allow_python_eval_exec(False)
 
-        coords = podpac.Coordinates([podpac.crange(-90, 90, 1.0), podpac.crange(-180, 180, 1.0)], dims=["lat", "lon"])
-        sine_node = SinCoords()
+            coords = podpac.Coordinates(
+                [podpac.crange(-90, 90, 1.0), podpac.crange(-180, 180, 1.0)], dims=["lat", "lon"]
+            )
+            sine_node = SinCoords()
 
-        with pytest.warns(UserWarning, match="Insecure evaluation"):
-            node = Arithmetic(A=sine_node, B=sine_node, eqn="2*abs(A) - B + {offset}", params={"offset": 1})
+            with pytest.warns(UserWarning, match="Insecure evaluation"):
+                node = Arithmetic(A=sine_node, B=sine_node, eqn="2*abs(A) - B + {offset}", params={"offset": 1})
 
-        with pytest.raises(PermissionError):
-            node.eval(coords)
+            with pytest.raises(PermissionError):
+                node.eval(coords)
 
     def test_missing_equation(self):
-        sine_node = SinCoords()
-        with pytest.raises(ValueError):
-            node = Arithmetic(A=sine_node, B=sine_node)
+        with podpac.settings:
+            podpac.settings.set_allow_python_eval_exec(True)
+
+            sine_node = SinCoords()
+            with pytest.raises(ValueError):
+                Arithmetic(A=sine_node, B=sine_node)
 
 
 class TestGeneric(object):
     def test_init_(self):
         a = SinCoords()
         b = Arange()
 
-        podpac.settings.set_allow_python_eval_exec(True)
-        node = Generic(code="import numpy as np\noutput = np.minimum(a,b)", a=a, b=b)
-
-        podpac.settings.set_allow_python_eval_exec(False)
-        with pytest.warns(UserWarning, match="Insecure evaluation"):
+        with podpac.settings:
+            podpac.settings.set_allow_python_eval_exec(True)
             node = Generic(code="import numpy as np\noutput = np.minimum(a,b)", a=a, b=b)
 
-    def test_evaluate(self):
-        podpac.settings.set_allow_python_eval_exec(True)
+            podpac.settings.set_allow_python_eval_exec(False)
```

Comment:
This actually creates a file... so probably better to set the settings['ALLOW_PYTHON_EVAL_EXEC']=True directly, based on how you wrote the context manager.